qa/tasks/cephfs/test_volumes.py

```diff
@@ -7888,3 +7888,43 @@ def test_unparseable_metafile_on_legacy_to_v1_upgrade(self):
 
         # remove group
         self._fs_cmd("subvolumegroup", "rm", self.volname, group)
+
+
+class TestEmptyStringForCreates(CephFSTestCase):
+    CLIENTS_REQUIRED = 1
+    MDSS_REQUIRED = 1
+
+    def setUp(self):
+        super().setUp()
+
+    def tearDown(self):
+        # clean up
+        super().tearDown()
+
+    def _fs_cmd(self, *args):
+        return self.mgr_cluster.mon_manager.raw_cluster_cmd("fs", *args)
+
+    def test_empty_name_string_for_subvolumegroup_name(self):
+        """
+        To test that an empty string is unacceptable for a subvolumegroup name
+        """
+        with self.assertRaises(CommandFailedError):
+            self._fs_cmd("subvolumegroup", "create", self.volname, "''")
+
+    def test_empty_name_string_for_subvolume_name(self):
+        """
+        To test that an empty string is unacceptable for a subvolume name
+        """
+        with self.assertRaises(CommandFailedError):
+            self._fs_cmd("subvolume", "create", "")
+
+    def test_empty_name_string_for_subvolumegroup_name_argument(self):
+        """
+        To test that an empty string is unacceptable for a subvolumegroup name
+        argument when creating a subvolume
+        """
+        with self.assertRaises(CommandFailedError):
+            self._fs_cmd("subvolume", "create", self.volname, "sv1", "--group_name")
+
+        with self.assertRaises(CommandFailedError):
+            self._fs_cmd("subvolume", "create", self.volname, "sv1", "--group_name", "''")
```

Comment:
Can't we add these test cases to their relevant classes? That way we can avoid creating a new class and the helper function that can be those classes too

Reply:
* creating a new class doesn't impose extra load during job run, so it should be okay
* the new class has been placed in `test_volumes.py` since the tests are relevant to volumes
* the `_fs_cmd()` function is used just as an abbreviation to the longer length function call that it uses; I could use the original function call as well
* a separate class will also help with fine grained testing for this specific feature while starting a suite run